types/src/validator_set.rs

```diff
@@ -84,7 +94,7 @@ impl CanonicalSerialize for ValidatorSet {
 
 impl CanonicalDeserialize for ValidatorSet {
     fn deserialize(deserializer: &mut impl CanonicalDeserializer) -> Result<Self> {
-        let size = deserializer.decode_u64()?;
+        let size = deserializer.decode_u32()?;
```

Comment:
We can probably just do `decode_vec` here

Reply:
There is a TODO to use `decode_vec` somewhere above, but I don't want to take that on in this diff. I only had to change this code because the recent diff that changed the vector serialization to use u32 lengths missed this case.